addon/models/node.js

```diff
@@ -105,5 +70,25 @@ export default OsfModel.extend({
     root: DS.belongsTo('node', {
         inverse: null
     }),
-    logs: DS.hasMany('logs')
+    logs: DS.hasMany('logs'),
+
+    save() {
+        // Some duplicate logic from osf-model#save needed to support
+        // contributor edits being saved through the node
+        var promise = this._super(...arguments);
+        var contributors = this.hasMany('contributors').hasManyRelationship;
+        if (contributors.hasData || contributors.hasLoaded) {
+            this.set(
+                '_dirtyRelationships.contributors.update',
+                contributors.members.list.filter(m => !m.record.get('isNew') && Object.keys(m.record.changedAttributes()).length > 0)
+            );
+            // Contributors are a 'real' delete, not just a de-reference
+            this.set(
+                '_dirtyRelationships.contributors.delete',
+                this.get('_dirtyRelationships.contributors.remove')
+            );
+            this.set('_dirtyRelationships.contributors.remove', []);
+        }
+        return promise;
```

Comment:
It seems like this function does two things: sending a request for the record, and also managing its relationships.

Is there a situation where one of those changes could fail, but not the other? Should we take that into account to ensure the state of data stays in sync?

For example: it seems we call `this.set` for everything, regardless of whether the promise succeeds.